### ISSUE TITLE
ShouldBeAbleToUseMultipleTheFirsts() unit test fails

### ISSUE THREAD

**Issue by lorddev** [nbuilder/nbuilder#62]
I discovered this while testing my changes re #55, and then checked out a bunch of other commits in history, and found that it was already failing, even in the master branch. I stepped through the code for like an hour and couldn't figure out the cause.

![screenclip](https://cloud.githubusercontent.com/assets/2220085/24384605/e1dd867c-1319-11e7-81b9-81d52fc7bd97.png)

**Comment by lorddev** [comment]
@crmckenzie Should documentation be updated in order to reflect this behavior? Also, why does Appveyor say the build passes if this test doesn't pass?

**Comment by crmckenzie** [comment]
@lorddev Tests pass for me. I'll have to dig into this at home. Not sure why you're getting different results.

**Comment by PureKrome** [comment]
Is this an issue of using a STATIC somewhere, so it becomes a poor race like condition?

I.e. @lorddev 's machine is faster??? or more parallel?? that @crmckenzie 's?  

Statics in tests cause havok :(

**Comment by PureKrome** [comment]
@lorddev does the test always fail for you when you run it BY ITSELF (and no other tests)?

**Comment by lorddev** [comment]
The test always fails, even if it's the only one running. (The method is not marked static.)

```

  Expected string length 16 but was 10. Strings differ at index 0.
  Expected: "OverwrittenTitle"
  But was:  "FirstTitle"
  -----------^

at FizzWare.NBuilder.Tests.Integration.ListBuilderTests_WithAClassThatHasAParameterlessConstructor.ShouldBeAbleToUseMultipleTheFirsts() in C:\projects\nbuilder\Source\FizzWare.NBuilder.Tests\Integration\ListBuilderTests_WithAClassThatHasAParameterlessConstructor.cs:line 281
```

**Comment by crmckenzie** [comment]
I just did a fresh pull on a separate machine and re-ran the tests in NBuilder-NETStandard16.sln.

![image](https://user-images.githubusercontent.com/947569/28648854-abb57c56-7225-11e7-908e-124f5a7918a1.png)

**Comment by PureKrome** [comment]
> The method is not marked static.

I didn't mean the test method, but a resource which is used, in the test.

>re-ran the tests in NBuilder-NETStandard16.sln

oh my! that's the Resharper runner!! I couldn't get my tests to run under that! I'm envious :(

**Comment by lorddev** [comment]
I use Resharper, too.

**Comment by PureKrome** [comment]
Any tricks into getting resharper to finding nunit tests?

**Comment by lorddev** [comment]
@PureKrome Have you tried Resharper -> Unit Tests -> Run all tests from solution ?

**Comment by PureKrome** [comment]
Yeah :( still nope :( It's just not finding/discovering them all in Resharper. I've also made sure that i've got nUnit discovery ON in Resharper.

this is the main reasons why :(

```
[30/07/2017 11:40:58 PM] Exploration skipped for <Fizzware.NBuilder-NetStandard16,NUnit 3x,.NETFramework,Version=v4.7>: project not supported by provider
[30/07/2017 11:40:58 PM] Exploration skipped for <Fizzware.NBuilder-NetStandard16,NUnit 2x,.NETFramework,Version=v4.7>: project not supported by provider
```

no idea what that means :(

logging debugger window output text...

```
[30/07/2017 11:40:58 PM] Exploration started for <Fizzware.NBuilder-NetStandard16,.NETFramework,Version=v4.7>
[30/07/2017 11:40:58 PM] Exploration started for <Fizzware.NBuilder-NetStandard16,.NETFramework,Version=v4.7> using <NUnit 3x,NUnitElementsSource> on thread <24:JetPool (S) Regular thread #1>
[30/07/2017 11:40:58 PM] Exploration skipped for <Fizzware.NBuilder-NetStandard16,NUnit 3x,.NETFramework,Version=v4.7>: project not supported by provider
[30/07/2017 11:40:58 PM] Exploration started for <Fizzware.NBuilder-NetStandard16,.NETFramework,Version=v4.7> using <NUnit 2x,NUnitElementsSource> on thread <24:JetPool (S) Regular thread #1>
[30/07/2017 11:40:58 PM] Exploration skipped for <Fizzware.NBuilder-NetStandard16,NUnit 2x,.NETFramework,Version=v4.7>: project not supported by provider
[30/07/2017 11:40:58 PM] Exploration started for <Fizzware.NBuilder-NetStandard16,.NETFramework,Version=v4.7> using <xUnit.net,XunitDotNetVsTestArtefactExplorer> on thread <24:JetPool (S) Regular thread #1>
[30/07/2017 11:40:58 PM] Exploration skipped for <Fizzware.NBuilder-NetStandard16,xUnit.net>: provider is disabled
[30/07/2017 11:40:58 PM] Exploration started for <Fizzware.NBuilder-NetStandard16,.NETFramework,Version=v4.7> using <MSTest,MsTestDotNetVsTestArtefactExplorer> on thread <24:JetPool (S) Regular thread #1>
[30/07/2017 11:40:58 PM] Exploration by <NUnit 3x> completed for <Fizzware.NBuilder-NetStandard16, .NETFramework,Version=v4.7>
[30/07/2017 11:40:58 PM] Exploration skipped for <Fizzware.NBuilder-NetStandard16,MSTest>: provider is disabled
[30/07/2017 11:40:58 PM] Exploration by <MSTest> completed for <Fizzware.NBuilder-NetStandard16, .NETFramework,Version=v4.7>
[30/07/2017 11:40:58 PM] Exploration by <xUnit.net> completed for <Fizzware.NBuilder-NetStandard16, .NETFramework,Version=v4.7>
[30/07/2017 11:40:58 PM] Exploration by <NUnit 2x> completed for <Fizzware.NBuilder-NetStandard16, .NETFramework,Version=v4.7>
[30/07/2017 11:40:58 PM] Exploration completed for <Fizzware.NBuilder-NetStandard16,.NETFramework,Version=v4.7>
[30/07/2017 11:40:58 PM] Exploration started for <Fizzware.NBuilder-NetStandard16,.NETStandard,Version=v1.6>
[30/07/2017 11:40:58 PM] Exploration completed for <Fizzware.NBuilder-NetStandard16,.NETStandard,Version=v1.6>
[30/07/2017 11:40:58 PM] Exploration started for <FizzWare.NBuilder.Tests,.NETCoreApp,Version=v1.1>
[30/07/2017 11:40:58 PM] Exploration started for <FizzWare.NBuilder.Tests,.NETCoreApp,Version=v1.1> using <xUnit.net,XunitDotNetVsTestArtefactExplorer> on thread <24:JetPool (S) Regular thread #1>
[30/07/2017 11:40:58 PM] Exploration skipped for <FizzWare.NBuilder.Tests,xUnit.net>: provider is disabled
[30/07/2017 11:40:58 PM] Exploration started for <FizzWare.NBuilder.Tests,.NETCoreApp,Version=v1.1> using <MSTest,MsTestDotNetVsTestArtefactExplorer> on thread <24:JetPool (S) Regular thread #1>
[30/07/2017 11:40:58 PM] Exploration by <xUnit.net> completed for <FizzWare.NBuilder.Tests, .NETCoreApp,Version=v1.1>
[30/07/2017 11:40:58 PM] Exploration skipped for <FizzWare.NBuilder.Tests,MSTest>: provider is disabled
[30/07/2017 11:40:58 PM] Exploration by <MSTest> completed for <FizzWare.NBuilder.Tests, .NETCoreApp,Version=v1.1>
[30/07/2017 11:40:58 PM] Exploration completed for <FizzWare.NBuilder.Tests,.NETCoreApp,Version=v1.1>
[30/07/2017 11:40:59 PM] Exploration started for <FizzWare.FunctionalTests.Model,.NETStandard,Version=v1.6>
[30/07/2017 11:40:59 PM] Exploration completed for <FizzWare.FunctionalTests.Model,.NETStandard,Version=v1.6>
```

**Comment by crmckenzie** [comment]
This PR was resolved in issue #55
